src/shared/components/post/post-form.tsx

```diff
@@ -73,19 +73,19 @@ interface PostFormState {
     community_id?: number;
     honeypot?: string;
   };
-  suggestedTitle?: string;
-  suggestedPosts?: PostView[];
-  crossPosts?: PostView[];
   loading: boolean;
+  suggestedPostsRes: RequestState<SearchResponse>;
+  metadataRes: RequestState<GetSiteMetadataResponse>;
```

Comment:
Move these two up to the `CreatePost` page, and pass in only the suggestedposts and suggestedtitle as props.

That means this will need an `onUrlChange(string...)`, that's received by the `CreatePost` page, so it can make those requests.

Reply:
I think it's worth passing the props in as `RequestState`s so that the form can tell when to show the loading spinner.